packages/theme-default/src/components/Search/logic/providers/LocalProvider.ts

```diff
@@ -3,14 +3,20 @@
  * Powered by FlexSearch. https://github.com/nextapps-de/flexsearch
  */
 
-import type { CreateOptions, Index as SearchIndex } from 'flexsearch';
-import FlexSearch from 'flexsearch';
+import type {
+  Document,
+  EnrichedDocumentSearchResultSetUnit,
+  IndexOptionsForDocumentSearch,
+} from 'flexsearch';
+import FlexSearchDocument from 'flexsearch/dist/module/document';
 import searchIndexHash from 'virtual-search-index-hash';
 import { type PageIndexInfo, SEARCH_INDEX_NAME } from '@rspress/shared';
 import type { SearchOptions } from '../types';
 import { LOCAL_INDEX, type Provider, type SearchQuery } from '../Provider';
 import { normalizeTextCase } from '../util';
 
+type TFlexSearchDocumentWithType = Document<PageIndexInfo, true>;
```

Comment:
The `T` prefix can be omitted, we have no convention to add a `T` prefix 😄

```suggestion
type FlexSearchDocumentWithType = Document<PageIndexInfo, true>;
```